src/state_machine/coordinator/mod.rs

```diff
@@ -744,42 +750,207 @@ pub mod test {
                 }
             }
-            _ => panic!("Expected Signature Operation result"),
+            OperationResult::SignSchnorr(sig) => {
+                if let SignatureType::Schnorr = signature_type {
+                    for coordinator in coordinators.iter() {
+                        assert!(sig.verify(
+                            &coordinator
+                                .get_aggregate_public_key()
+                                .expect("No aggregate public key set!")
+                                .x(),
+                            msg
+                        ));
+                        assert_eq!(coordinator.get_state(), State::Idle);
+                    }
+                } else {
+                    panic!("Expected OperationResult::SignSchnorr");
+                }
+            }
+            OperationResult::SignTaproot(sig) => {
+                if let SignatureType::Taproot(merkle_root) = signature_type {
+                    for coordinator in coordinators.iter() {
+                        let tweaked_public_key = compute::tweaked_public_key(
+                            &coordinator
+                                .get_aggregate_public_key()
+                                .expect("No aggregate public key set!"),
+                            merkle_root,
+                        );
+
+                        assert!(sig.verify(&tweaked_public_key.x(), msg));
+                        assert_eq!(coordinator.get_state(), State::Idle);
+                    }
+                } else {
+                    panic!("Expected OperationResult::SignTaproot");
+                }
+            }
+            _ => panic!("Expected OperationResult"),
+        }
+
+        operation_results[0].clone()
+    }
+
+    pub fn run_dkg_sign<Coordinator: CoordinatorTrait, SignerType: SignerTrait>(
+        num_signers: u32,
+        keys_per_signer: u32,
+    ) {
+        let (mut coordinators, mut signers) =
+            run_dkg::<Coordinator, SignerType>(num_signers, keys_per_signer);
+
+        let msg = "It was many and many a year ago, in a kingdom by the sea"
+            .as_bytes()
+            .to_vec();
+
+        run_sign::<Coordinator, SignerType>(
+            &mut coordinators,
+            &mut signers,
+            &msg,
+            SignatureType::Frost,
+        );
+        run_sign::<Coordinator, SignerType>(
+            &mut coordinators,
+            &mut signers,
+            &msg,
+            SignatureType::Schnorr,
+        );
+        run_sign::<Coordinator, SignerType>(
+            &mut coordinators,
+            &mut signers,
+            &msg,
+            SignatureType::Taproot(None),
+        );
+        run_sign::<Coordinator, SignerType>(
+            &mut coordinators,
+            &mut signers,
+            &msg,
+            SignatureType::Taproot(Some([128u8; 32])),
+        );
+    }
+
+    pub fn check_signature_shares<Coordinator: CoordinatorTrait, SignerType: SignerTrait>(
```

Comment:
Can we add a description of what this function is supposed to check. It looks like it checks that we can run dkg and get a signing error under some specific circumstance but I couldn't quite figure out what triggers the error.

Reply:
Done